src/resource.rs

```diff
@@ -534,11 +546,11 @@ mod tests {
             >,
         > {
             web::resource("/test-compat")
-                // .wrap_fn(|req, srv| {
-                //     let fut = srv.call(req);
-                //     async { Ok(fut.await?.map_into_right_body::<()>()) }
-                // })
-                .wrap(Compat::new(DefaultHeaders::new()))
+                .wrap_fn(|req, srv| {
+                    let fut = srv.call(req);
+                    async { Ok(fut.await?.map_into_right_body::<()>()) }
+                })
+                .wrap(Compat::noop())
```

Comment:
I think we should encourage returning `impl HttpServiceFactory` for `Scope` and `Resource`.

Reply:
I agree, will add it in. I included this test to show that the changes to not interfere with these possible-in-the-wild signatures.